tests/suites/options.js

```diff
@@ -641,3 +641,54 @@ test('Multidate Separator', function(){
     target.click();
     equal(input.val(), '2012-03-05 2012-03-04 2012-03-12');
 });
+
+
+test("Picker is shown on input focus when showOnFocus is not defined", function () {
+
+    var input = $('<input />')
+            .appendTo('#qunit-fixture')
+            .val('2014-01-01')
+            .datepicker({
+            }),
+        dp = input.data('datepicker'),
+        picker = dp.picker;
+
+    input.focus();
+
+    ok(picker.is(":visible"), "Datepicker is hidden");
+
+});
+
+test("Picker is shown on input focus when showOnFocus is true", function () {
+
+    var input = $('<input />')
+            .appendTo('#qunit-fixture')
+            .val('2014-01-01')
+            .datepicker({
+                showOnFocus: true
+            }),
+        dp = input.data('datepicker'),
+        picker = dp.picker;
+
+    input.focus();
+
+    ok(picker.is(":visible"), "Datepicker is hidden");
+
+});
+
+test("Picker is hidden on input focus when showOnFocus is false", function () {
+
+    var input = $('<input />')
+            .appendTo('#qunit-fixture')
+            .val('2014-01-01')
+            .datepicker({
+                showOnFocus: false
+            }),
+        dp = input.data('datepicker'),
+        picker = dp.picker;
+
+    input.focus();
+
+    ok(picker.is(":hidden"), "Datepicker is visible");
```

Comment:
I'm also getting to know the qunit framework, but as far as I can see the message you can give (optional) to the ok method, is a message to show when the test is done (instead of the result). See [qunit::ok docs](http://api.qunitjs.com/ok/), they describe the message as "A short description of the assertion"

So i guess the messages have to change